docs/source/nested.rst

```diff
@@ -39,31 +39,78 @@ nested_tensor([
   tensor([3., 4., 5., 6., 7.], device='cuda:0')
 ])
 
+In order to form a valid NestedTensor the passed Tensors also all need to match in dimension, but none of the other attributes need to.
+
+>>> a = torch.randn(3, 50, 70) # image 1
+>>> b = torch.randn(3, 128, 64) # image 2
+>>> nt = torch.nested_tensor([a, b], dtype=torch.float32)
+>>> nt.dim()
+4
+
+If one of the dimensions don't match, the constructor throws an error.
+
+>>> a = torch.randn(50, 128) # text 1
+>>> b = torch.randn(3, 128, 64) # image 2
+>>> nt = torch.nested_tensor([a, b], dtype=torch.float32)
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+RuntimeError: All Tensors given to nested_tensor must have the same dimension. Found dimension 3 for Tensor at index 1 and dimension 2 for Tensor at index 0.
+
 Note that the passed Tensors are being copied into a contiguous piece of memory. The resulting
 NestedTensor allocates new memory to store them and does not keep a reference.
 
-In order to form a valid NestedTensor the passed Tensors also all need to match in dimension.
 
 At this moment we only support one level of nesting, i.e. a simple, flat list of Tensors. In the future
 we can add support for multiple levels of nesting, such as a list that consists entirely of lists of Tensors.
 Note that for this extension it is important to maintain an even level of nesting across entries so that the resulting NestedTensor
 has a well defined dimension. If you have a need for this feature, please feel encourage to open a feature request so that
 we can track it and plan accordingly.
 
+size
++++++++++++++++++++++++++
+
+Even though a NestedTensor does not support .size(), it supports the size along a given dimension if that dimension is regular.
+
+>>> a = torch.randn(50, 128) # text 1
+>>> b = torch.randn(32, 128) # text 2
+>>> nt = torch.nested_tensor([a, b], dtype=torch.float32)
+>>> nt.size(0)
```

Comment:
also, is size function usage still recommended? at some point i had a feeling that shape is now recommended over older size

Reply:
well, I meant shape[idx] of course

Reply:
The problem with `.shape` is that it returns a full `torch.Size`. While `shape[idx]` as an expression will yield a valid result, the result of `.shape` is not defined. `.size(i)` effectively provides a fusion of `.shape` + `__getitem__`.